frame/transaction-payment/src/lib.rs

```diff
@@ -217,7 +227,7 @@ where
 
 		if positive {
 			let excess = first_term.saturating_add(second_term).saturating_mul(previous);
-			previous.saturating_add(excess).max(min_multiplier)
+			previous.saturating_add(excess).max(M::get()).min(X::get())
```

Comment:
I'd rather put these two in a let binding and use a better identifier:

```suggestion
			previous.saturating_add(excess).max(min_multiplier).min(max_multiplier)
```